src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -236,7 +236,7 @@ private BoundExpression FixTupleLiteral(ArrayBuilder<DeconstructionVariable> che
             if (type.IsTupleType)
             {
                 // tuple literal such as `(1, 2)`, `(null, null)`, `(x.P, y.M())`
-                tupleOrDeconstructedTypes = type.TupleElementTypes;
+                tupleOrDeconstructedTypes = type.TupleElementTypes.SelectAsArray(TypeMap.AsTypeSymbol);
```

Comment:
If you don't have them already, could you test:
```c#
(var x, var y) = ((string?)null, "hello");
x = null; // ok
y = null; // warn
```
- with a method returning a tuple `(var x, var y) = M(); // where (string?, string) M()`.
- with a Deconstruct method `void Deconstruct(out string? x, out string y)`.
- it's probably good to test one of the above in a deconstruction foreach too `foreach ((var x, var y) in ...)`.
Thanks

Reply:
Great suggestions, thanks. Added.